Docstring update in polysys

mijo2 opened the pull request:
#### References to other Issues or PRs
Updated some of the docstrings of public functions in solvers/polysys.py in accordance with #15370  


#### Brief description of what is fixed or changed
--> Updated docstrings of public functions in solvers/polysys.py

#### Other comments


#### Release Notes

<!-- BEGIN RELEASE NOTES -->
* solvers
  * Updated docstring of public functions in polysys
<!-- END RELEASE NOTES -->

mijo2 committed:
solvers: Updated docstrings in polysys

Some public functions in polysys don't follow numpy docstring
format. So, the doc strings were updated accordingly.

smichr commented on sympy/solvers/polysys.py:
```diff
@@ -20,6 +20,24 @@ def solve_poly_system(seq, *gens, **args):
+
+    seq: a list/tuple/set
+        Listing all the equations that are needed to be solved
+    *gens: symbols
```
I think `gens` is more general than symbols, e.g. maybe it can be any generator of the poly like `cos(x)` in `Poly(cos(x)-1)`.

mijo2 replied:
@smichr Sure I will update it.

mijo2 committed:
solvers: *gens is for generators

Initially, in the docstring, the *gens were listed as symbols in
the docstring but *gens means generators.

sylee957 commented on sympy/solvers/polysys.py:
```diff
@@ -20,6 +20,24 @@ def solve_poly_system(seq, *gens, **args):
+    Returns
+    =======
+
+    A List of tuples
```
This should better be `List[Tuple]` or `List` and put the detailed description about the type in the indented block below.

mijo2 replied:
@sylee957 You mean something like below:

```
List[Tuple]
    A list of tuples
    Solutions for symbols that satisfy the equations listed
    in polys
```

sylee957 commented on sympy/solvers/polysys.py:
```diff
@@ -20,6 +20,24 @@ def solve_poly_system(seq, *gens, **args):
+    *gens: generators
+        generators of the equations in seq for which we want the
+        solutions
+    **args: Keyword arguments
```
I think that stars should be removed.

mijo2 replied:
@sylee957 I will make the necessary changes.

mijo2 committed:
solvers: Corrected some docstring

*gens and **args were updated to gens and args in the docstring.
Along with that, some docstring in solve_generic was corrected.
